controllers/storagecluster/cephcluster.go

```diff
@@ -266,6 +266,14 @@ func (obj *ocsCephCluster) ensureCreated(r *StorageClusterReconciler, sc *ocsv1.
 		return reconcile.Result{}, nil
 	}
 
+	if found.Status.CephStorage != nil {
+		cephDeviceClasses := []string{}
+		for _, deviceClass := range found.Status.CephStorage.DeviceClasses {
+			cephDeviceClasses = append(cephDeviceClasses, deviceClass.Name)
+		}
+		sc.Status.CephDeviceClasses = cephDeviceClasses
```

Comment:
Why are the device classes duplicated in the StorageCluster CR status? Can the CephCluster status be used directly instead?

Reply:
Hm, actually there is no way to pass the cephcluster status to the generateDeviceClass function, So either we have to use a package level var to share the information or do this. Also keeping the status on the storagecluster status make it possible to write unit tests for this, so I thought of this approach.

Reply:
Anytime a CRD spec or status exposes information it is considered API that users may consume. We don't want to put info in the status as an internal implementation detail if the user never needs to know it. In other words, let's avoid exposing unnecessary API information.

Something that may be useful for the user to be aware of, is which is the default device class name. So instead of including all device classes in this StorageCluster status to evaluate later during reconcile, how about we immediately evaluate the default storage class name and set it here? We have all the parameters needed at this point to determine the default storage class, right?

Reply:
Thanks for the suggestions, I have updated the PR based on this. Do have another look.